File: C/ex3/predictOneVsAll.c
#include <stdio.h>
double sigmoid(double z);

int predictOneVsAll(int M, int N, double all_theta[10][N], double X[M][N], double g[M])
{
        double TX;
        int i, j, k;
	  double gg;
	  double max;

        for(i=0; i < M; i++){
	     g[i] = 0;
	     max = -1;
	     
	     for(k=0; k < 10; k++){
	         TX = 1;
              for(j=0; j < N; j++)
                 TX += X[i][j] * all_theta[k][j];
		     
               gg = sigmoid(TX);
		   
		   if(gg >  max ){
			max = gg;
		      g[i] = k;
			if(g[i] == 0)g[i]=10;      
		  }
	     }
	     	     
        }	  

        return 0 ;
}

File: Fortran/ex3/predictOneVsAll.f90
	subroutine predictOneVsAll(all_theta, g)
	use mod_common
	
	implicit none 
	double precision,dimension(M)::g
	double precision::sig
	double precision,dimension(num_labels,N)::all_theta
	double precision::TX, max, sigmoid
	integer::i, k, j
	
	do i=1, M
	       max = -1
		 g(i)= 0
		 
	      do k=1, num_labels
	          TX = 1
		    do j=1, N
			TX = TX + X(i,j)*all_theta(k,j)
		    enddo
		    
		    sig = sigmoid(TX)
		    
		    if(sig.gt.max)then
		       max = sig
			 g(i) = k
			 if(g(i).eq.0)g(i)=10.
		   endif
		    
		enddo
								 
	enddo

		
	 end subroutine predictOneVsAll
